aries_vcx/src/libindy/utils/signus.rs

```diff
@@ -17,3 +21,95 @@ pub async fn create_and_store_my_did(seed: Option<&str>, method_name: Option<&st
         .map_err(VcxError::from);
     res
 }
+
+pub async fn rotate_verkey(did: &str) -> VcxResult<()> {
+    let trustee_temp_verkey = libindy_replace_keys_start(did).await?;
+    let nym_request = ledger::libindy_build_nym_request(&did, &did, Some(&trustee_temp_verkey), None, None).await?;
+    let nym_result = ledger::libindy_sign_and_submit_request(&did, &nym_request).await?;
+    let nym_result_json: Value = serde_json::from_str(&nym_result)
+        .map_err(|err| VcxError::from_msg(VcxErrorKind::SerializationError, format!("Cannot deserialize {:?} into Value, err: {:?}", nym_result, err)))?;
+    let response_type: String = nym_result_json["op"].as_str()
+        .ok_or(VcxError::from_msg(VcxErrorKind::SerializationError, format!("Cannot failed to convert {:?} into str", nym_result_json["op"])))?.to_string();
+    if response_type != "REPLY" {
+        return Err(VcxError::from_msg(VcxErrorKind::InvalidLedgerResponse, format!("Obained non-success ledger response: {}", nym_result_json)));
+    }
+    libindy_replace_keys_apply(&did).await
+}
+
+pub async fn libindy_replace_keys_start(did: &str) -> VcxResult<String> {
+    if PoolMocks::has_pool_mock_responses() {
+        warn!("libindy_replace_keys_start >> retrieving pool mock response");
+        Ok(PoolMocks::get_next_pool_response())
+    } else {
+        did::replace_keys_start(get_wallet_handle(), did, "{}")
+            .map_err(VcxError::from)
+            .await
+    }
+}
+
+pub async fn libindy_replace_keys_apply(did: &str) -> VcxResult<()> {
+    if pool_mocks_enabled() {
+        warn!("libindy_replace_keys_apply >> retrieving pool mock response");
+        Ok(())
+    } else {
+        did::replace_keys_apply(get_wallet_handle(), did)
+            .map_err(VcxError::from)
+            .await
+    }
+}
+
+pub async fn get_verkey_from_wallet(did: &str) -> VcxResult<String> {
+    if PoolMocks::has_pool_mock_responses() {
+        warn!("get_verkey_from_wallet >> retrieving pool mock response");
+        Ok(PoolMocks::get_next_pool_response())
+    } else {
+        did::key_for_local_did(get_wallet_handle(), did)
+            .map_err(VcxError::from)
+            .await
+    }
+}
+
+pub async fn get_verkey_from_ledger(did: &str) -> VcxResult<String> {
```

Comment:
We could  later experimentally try `indy-vdr` for ledger calls to avoid parsing like bellow, as alternative to `vdr-tools` (esp. since doesn't provide native rust api)

Reply:
but not a priority now